cpp/ccugraph_alg_module/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+set(ccugraph_alg_module_src
+    main.cu)
+
+add_query_module(ccugraph_alg 1 "${ccugraph_alg_module_src}")
+
+# TODO(gitbuda): Add macro for cuda set_target_properties + deal with CUDA_ARCHITECTURES somehow.
```

Comment:
Consider adding all that as a part of `add_cugraph_query_module`, e.g. specific macros like `set_cuda_target_properties` can be avoided.